dataflow/src/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -478,7 +478,7 @@ public boolean equals(Object obj) {
                 return false;
             }
             LocalVariable other = (LocalVariable) obj;
-            return other.toString().equals(toString());
+            return other.element.equals(element);
```

Comment:
I already made this change in the mainline. Not sure why it didn't merge as I expected.